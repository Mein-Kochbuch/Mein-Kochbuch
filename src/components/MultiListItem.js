import React from "react";
import styled from "styled-components/native/dist/styled-components.native.esm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function MultiListItem({title}) {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <StyledTest>
                {title}
            </StyledTest>
        </StyledTouchableOpacity>
    )
}

const StyledTest = styled.Text`
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 10px;
  margin: 12px;
  display: flex;
  align-items: center;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
