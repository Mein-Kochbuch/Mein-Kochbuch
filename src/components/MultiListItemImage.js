import Icon from "react-native-vector-icons/Feather";
import React from "react";
import styled from "styled-components/native";

export default function MultiListItemImage({icon}) {

    return (
        icon === "heart" ? <Icon name={"heart"} size={100} color={"#FF0000"}/> :
            <StyledImage source={require("../../resources/platzhalter.png")}/>
    )
}

const StyledImage = styled.Image`
  height: 100px;
  width: 100px;

  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`
