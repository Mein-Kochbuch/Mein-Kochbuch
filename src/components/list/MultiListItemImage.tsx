import Icon from 'react-native-vector-icons/Feather';
import React from 'react';
import styled from 'styled-components/native';

interface MultiListItemImageProps {
  icon: string;
}

export default function MultiListItemImage({icon}: MultiListItemImageProps) {
  return icon === 'heart' ? (
    <Icon name={'heart'} size={100} color={'#FF0000'} />
  ) : (
    <StyledImage source={require('../../../resources/platzhalter.png')} />
  );
}

const StyledImage = styled.Image`
  height: 100px;
  width: 100px;

  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`;
