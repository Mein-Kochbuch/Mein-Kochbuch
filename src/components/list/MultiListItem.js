import React from "react";
import styled from "styled-components/native/dist/styled-components.native.esm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";
import MultiListItemImage from "./MultiListItemImage";

export default function MultiListItem({title, icon}) {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <MultiListItemImage icon={icon}/>
            <StyledTest>
                {title}
            </StyledTest>
        </StyledTouchableOpacity>
    )
}

const StyledTest = styled.Text`
  text-align: center;
  font-size: 30px;
  margin-left: 12px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 10px;
  margin: 12px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
