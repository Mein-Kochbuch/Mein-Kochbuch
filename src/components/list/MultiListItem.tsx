import React from 'react';
import styled from 'styled-components/native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import {Appearance} from 'react-native';
import MultiListItemImage from './MultiListItemImage';

interface MultiListItemProps {
  title: string;
  icon: string;
}

export default function MultiListItem({title, icon}: MultiListItemProps) {
  const isDarkMode = Appearance.getColorScheme() === 'dark';

  return (
    <StyledTouchableOpacity isDarkMode={isDarkMode}>
      <MultiListItemImage icon={icon} />
      <StyledTest>{title}</StyledTest>
    </StyledTouchableOpacity>
  );
}

const StyledTest = styled.Text`
  text-align: center;
  font-size: 30px;
  margin-left: 12px;
`;

interface StyledTouchableopacityProps {
  isDarkMode: boolean;
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableopacityProps>`
  padding: 10px;
  margin: 12px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-style: solid;
  border-color: ${props => (props.isDarkMode ? Colors.lighter : Colors.darker)};
  border-radius: 10px;
  border-width: 2px;
`;
