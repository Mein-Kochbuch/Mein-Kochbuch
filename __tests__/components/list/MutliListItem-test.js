import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MultiListItem from "../../../src/components/list/MultiListItem";
import MultiListItemImage from "../../../src/components/list/MultiListItemImage";
import {useColorScheme} from "react-native-appearance";

jest.mock("react-native-appearance", () => ({
    useColorScheme: jest.fn(),
}));

jest.mock("../../src/components/MultiListItemImage", () => {
    return (props) => {
        return <div {...props}/>
    }
});

describe("MultiListItem Test", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('dark', () => {
        useColorScheme.mockReturnValueOnce('dark');

        const component = renderer.create(
            <MultiListItem title={"test-title"} icon={"test-icon"}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
        expect(component.toJSON().children.length).toBe(2)
        expect(component.toJSON().children[1].children[0]).toBe("test-title")

        const testInstance = component.root

        expect(testInstance.findAllByType(MultiListItemImage)[0].props.icon).toBe("test-icon")
    })

    it('dark', () => {
        useColorScheme.mockReturnValueOnce('light');

        const component = renderer.create(
            <MultiListItem title={"test-title"} icon={"test-icon"}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
        expect(component.toJSON().children.length).toBe(2)
        expect(component.toJSON().children[1].children[0]).toBe("test-title")

        const testInstance = component.root

        expect(testInstance.findAllByType(MultiListItemImage)[0].props.icon).toBe("test-icon")
    })
})
