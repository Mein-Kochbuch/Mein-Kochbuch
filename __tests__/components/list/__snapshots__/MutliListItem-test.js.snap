// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiListItem Test dark 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#F3F3F3",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 12,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 12,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <div
    icon="test-icon"
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "marginLeft": 12,
          "textAlign": "center",
        },
      ]
    }
  >
    test-title
  </Text>
</View>
`;

exports[`MultiListItem Test light 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#222",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 12,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 12,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <div
    icon="test-icon"
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "marginLeft": 12,
          "textAlign": "center",
        },
      ]
    }
  >
    test-title
  </Text>
</View>
`;
